src/Lykke.Payments.Link4Pay/Controllers/WebhookController.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Common;
+using Common.Log;
+using Lykke.Common.Log;
+using Lykke.Cqrs;
+using Lykke.Payments.Link4Pay.Contract;
+using Lykke.Payments.Link4Pay.Domain;
+using Lykke.Payments.Link4Pay.Domain.Repositories.RawLog;
+using Lykke.Payments.Link4Pay.Domain.Services;
+using Lykke.Payments.Link4Pay.Workflow.Commands;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+
+namespace Lykke.Payments.Link4Pay.Controllers
+{
+    public class WebhookController : Controller
+    {
+        private readonly ILink4PayApiService _link4PayApiService;
+        private readonly IEncryptionService _encryptionService;
+        private readonly IRawLogRepository _rawLogRepository;
+        private readonly ICqrsEngine _cqrsEngine;
+        private readonly ILog _log;
+
+        public WebhookController(
+            ILink4PayApiService link4PayApiService,
+            IEncryptionService encryptionService,
+            IRawLogRepository rawLogRepository,
+            ICqrsEngine cqrsEngine,
+            ILogFactory logFactory
+            )
+        {
+            _link4PayApiService = link4PayApiService;
+            _encryptionService = encryptionService;
+            _rawLogRepository = rawLogRepository;
+            _cqrsEngine = cqrsEngine;
+            _log = logFactory.CreateLog(this);
+        }
+
+        [HttpPost]
+        [Route("/webhook")]
+        public async Task<ActionResult> Webhook()
```

Comment:
How will it be hosted? Will it be whitelist for link4pay? Will not it be possible to call from outside by third-party?

Reply:
no, the payload is encrypted